lib/mongoid/criteria.rb

```diff
@@ -55,6 +57,40 @@ def ==(other)
       entries == other
     end
 
+    # Call either Findable#find or Enumerable#find based on the given arguments
+    #
+    # If the arguments match Enumerable#find signature, that will be invoked.
+    # Otherwise Findable#find will be invoked
+    #
+    # The signature of Enumerable#find:
+    #
+    #   find(ifnone = nil) { |obj| block } → obj or nil
+    #   find(ifnone = nil) → an_enumerator
+    #
+    # @example Find using a block, invokes Enumerable#find
+    #   criteria.find { |item| item.name == "Depeche Mode" }
+    #
+    # @example Find given a block and a Proc, invokes Enumerable#find
+    #   criteria.find(-> { "Default Band" }) { |item| item.name == "Milwaukee Mode" }
```

Comment:
Is there actual usage of this pattern in the wild, considering one can just as easily say

<pre>
     #   criteria.find { |item| item.name == "Milwaukee Mode" } || 'Default Band'
</pre>

Reply:
Yeah, I've never seen it. I was just trying to cover all the cases for `Enumerable#find` :)